projects/storefrontlib/src/cms-components/configurator/commons/config-form/config-form.component.ts

```diff
@@ -31,35 +27,25 @@ export class ConfigFormComponent implements OnInit, OnDestroy {
   ) {}
 
   ngOnInit(): void {
-    this.subscription.add(
-      this.routingService
-        .getRouterState()
-        .subscribe(state => this.initConfigurationForm(state))
+    this.configuration$ = this.routingService.getRouterState().pipe(
+      map(routingData => routingData.state.params.rootProduct),
+      switchMap(product =>
+        this.configuratorCommonsService.getConfiguration(product)
+      ),
+      tap(
```

Comment:
Can we not initialize the stream `this.currentGroup$` in the tap?
Having a separate definition of the stream deriving from the `this.configuration$` could be cleaner. for example:
```typescript
this.currentGroup$ = this.configuration$.pipe(
  switchMap(configuration => this.configuratorGroupsService.getCurrentGroup(
            configuration.productCode
          )
   ));
```

Reply:
I see...Yes, will change this (in a different branch feature/GH-5271 though)